### PR TITLE
C350712 Check date and time --fee/fines (firebird) (TaaS)

### PR DESCRIPTION
**Pull request by IlyaLyskavets**
[Test rail](https://foliotest.testrail.io/index.php?/cases/view/350712)

[Allure](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4593/allure/)

### CODE REVIEW

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import uuid from 'uuid';
+import moment from 'moment';
+import { TestTypes } from '../../support/dictionary';
+import devTeams from '../../support/dictionary/devTeams';
+import permissions from '../../support/dictionary/permissions';
+import servicePoints from '../../support/fragments/settings/tenant/servicePoints/servicePoints';
```
Comment (Scorpibear) [inline cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js:6]: Use PascalCase for imported fragments to match the code style

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import uuid from 'uuid';
+import moment from 'moment';
+import { TestTypes } from '../../support/dictionary';
```
[the remark this answers is not in the document]
Comment (IlyaLyskavets) [reply cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js]: Sure, done

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js (hunk 1 above)
Comment (IlyaLyskavets) [reply cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js]: Got it, done

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          paymentMethods.name = name;
+          paymentMethods.id = id;
+        });
+
+        cy.createTempUser([
+          permissions.uiUsersView.gui,
+          permissions.uiUsersfeefinesCRUD.gui,
+          permissions.uiUsersViewServicePoints.gui,
+          permissions.uiUsersfeefinesView.gui,
+          permissions.uiUsersManualCharge.gui,
```
[the remark this answers is not in the document]
Comment (IlyaLyskavets) [reply cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js]: Done accordingly to the test case

File: cypress/support/fragments/circulation-log/searchPane.js [hunk 4]
```diff
@@ -235,4 +235,11 @@ export default {
     SearchResults.clickOnCell(barcode, 0);
     ItemRecordView.waitLoading();
   },
+
+  searchByBilled() {
+    cy.do([
+      Accordion({ id: 'fee' }).clickHeader(),
+      Checkbox({ id: 'clickable-filter-fee-billed' }).click(),
+    ]);
```
Comment (nayimovag) [inline cypress/support/fragments/circulation-log/searchPane.js:243]: use this method setFilterOptionFromAccordion

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        });
+        PaymentMethods.createViaApi(ownerData.id).then(({ name, id }) => {
+          PaymentMethods.name = name;
+          PaymentMethods.id = id;
+        });
+
+        cy.createTempUser([
+          Permissions.circulationLogAll.gui,
+          Permissions.uiUsersfeefinesView.gui,
+          Permissions.uiUsersfeefinesCRUD.gui,
```
Comment (nayimovag) [inline cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js:54]: do you need crud permissions?

File: cypress/support/fragments/users/feeFineDetails.js [hunk 6]
```diff
@@ -18,4 +18,8 @@ export default {
   confirmFeeFineCancellation: (comment) => {
     cy.do([TextArea({ name: 'comment' }).fillIn(comment), Button({ type: 'submit' }).click()]);
   },
+  verifyBilledDateValue: (expectedBilledDateValue) => {
+    const billedDateLabel = 'Billed date';
+    cy.expect(KeyValue(billedDateLabel).has({ value: expectedBilledDateValue }));
```
Comment (nayimovag) [inline cypress/support/fragments/users/feeFineDetails.js:23]: either save whole locator to variable or move label name to expect assertion

**Review by nayimovag — CHANGES_REQUESTED** [review]

File: cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js (hunk 5 above)
Comment (IlyaLyskavets) [reply cypress/e2e/circulation-log/same-date-and-time-format-as-fee-fine-details-page.cy.js:54]: Apparently, Select 'Fee/fine Details' option from Actions dropdown is visible only when I use crud permission
Without CRUD: 
image
With CRUD: 
image

File: cypress/support/fragments/circulation-log/searchPane.js (hunk 4 above)
Comment (IlyaLyskavets) [reply cypress/support/fragments/circulation-log/searchPane.js]: done

**Review by nayimovag — APPROVED** [review]

**Review by Scorpibear — APPROVED** [review]

**Pull request merged**